consumer.go

```diff
@@ -171,9 +221,19 @@ func (c *consumer) sync(force bool) (*ConsumerSync, error) {
 	}
 
 	// open remote reader
-	reader, err := NewReader(c.ctx, c.remote, &c.opt.ReaderOptions)
-	if err != nil {
-		return nil, err
+	var reader *Reader
+	if c.opt.incremental {
+		var remotes []*bfs.Object
+		if reader, remotes, err = c.newIncrementalReader(); err != nil {
```

Comment:
that's not good. the remotes should be embedded into `reader` and `reader.Close()` should close those remotes. no need to return them here